From 4dda2d06f4a4c9a526ccdac389ab3c51475fb6b0 Mon Sep 17 00:00:00 2001
Message-Id: <4dda2d06f4a4c9a526ccdac389ab3c51475fb6b0@dist-git>
From: =?UTF-8?q?J=C3=A1n=20Tomko?= <jtomko@redhat.com>
Date: Tue, 2 Dec 2014 16:12:34 +0100
Subject: [PATCH] Silently ignore MAC in NetworkLoadConfig

Libvirt's RPMs have been adding it to networks which don't support it.

https://bugzilla.redhat.com/show_bug.cgi?id=1156367
(cherry picked from commit c9c7a2bd964d02f1c1ffc542cb18bf851852dd37)

Signed-off-by: Jiri Denemark <jdenemar@redhat.com>
---
 src/conf/network_conf.c | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/src/conf/network_conf.c b/src/conf/network_conf.c
index 863d0e2..d348a20 100644
--- a/src/conf/network_conf.c
+++ b/src/conf/network_conf.c
@@ -3152,6 +3152,10 @@ virNetworkObjPtr virNetworkLoadConfig(virNetworkObjListPtr nets,
          */
         if (virNetworkSetBridgeName(nets, def, 0))
             goto error;
+    } else {
+        /* Throw away MAC address for other forward types,
+         * which could have been generated by older libvirt RPMs */
+        def->mac_specified = false;
     }
 
     if (!(net = virNetworkAssignDef(nets, def, false)))
-- 
2.2.0

